src/screens/MarketDetailScreen/MarketDetailPage.tsx

```diff
@@ -0,0 +1,242 @@
+import React, {useRef, useState, useCallback} from 'react';
+import {
+  Alert,
+  TouchableOpacity,
+  View,
+  ScrollView,
+  NativeSyntheticEvent,
+  NativeScrollEvent,
+  LayoutChangeEvent,
+} from 'react-native';
+import date from '@utils/date';
+import Menu from '@/components/marketDetailPage/Menu';
+import S from './MarketDetail.style';
+import {MarketType} from '@/types/Market';
+import {ProductType} from '@/types/ProductType';
+
+//TODO: 장바구니 타입 확정 후 네비게이션
+
+type CartItem = {
+  productId: number;
+  productName: string;
+  count: number;
+};
+
+const MarketDetailPage = ({detail}: {detail: MarketType}) => {
+  const [cart, setCart] = useState<CartItem[]>([]);
+  const [selectedTag, setSelectedTag] = useState<string>('추천메뉴');
+  const scrollViewRef = useRef<ScrollView>(null);
+  const tagScrollViewRef = useRef<ScrollView>(null);
+  const [sectionOffsets, setSectionOffsets] = useState<{[key: string]: number}>(
+    {},
+  );
+  const [sectionHeights, setSectionHeights] = useState<{[key: string]: number}>(
+    {},
+  );
+  const [tagWidths, setTagWidths] = useState<{[key: string]: number}>({});
+  const {name, pickupStartAt, pickupEndAt, address, products} = detail;
```

Comment:
```suggestion
const MarketDetailPage = ({name, pickupStartAt, pickupEndAt, address, products}: MarketType) => {
  const [cart, setCart] = useState<CartItem[]>([]);
  const [selectedTag, setSelectedTag] = useState<string>('추천메뉴');
  const scrollViewRef = useRef<ScrollView>(null);
  const tagScrollViewRef = useRef<ScrollView>(null);
  const [sectionOffsets, setSectionOffsets] = useState<{[key: string]: number}>(
    {},
  );
  const [sectionHeights, setSectionHeights] = useState<{[key: string]: number}>(
    {},
  );
  const [tagWidths, setTagWidths] = useState<{[key: string]: number}>({});
```

props를 컴포넌트 내부에서 분해 할당하는 대신 props의 타입에서 분해 할당하면 다른 컴포넌트와 컨벤션을 맞출 수 있겠습니다

Reply:
props를 detail로 받아왔던 방식을 수정해서 다른 컴포넌트랑 컨벤션을 맞추긴 했는데 어느방식이 나은진 모르겠네요 ㅎㅎ 우선 반영했습니다

Reply:
[vercel/nextjs](https://github.com/vercel/next.js/blob/canary/packages/next/src/client/components/app-router.tsx#L630)

```typescript
export default function AppRouter({
  actionQueue,
  globalErrorComponent,
  assetPrefix,
}: {
  actionQueue: AppRouterActionQueue
  globalErrorComponent: ErrorComponent
  assetPrefix: string
}) {
```

[reactjs/react.dev](https://github.com/reactjs/react.dev/blob/main/src/components/Button.tsx#L8)
```typescript
interface ButtonProps {
  children: React.ReactNode;
  onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
  active?: boolean;
  className?: string;
  style?: Record<string, string>;
}

export function Button({
  children,
  onClick,
  active = false,
  className,
  style,
}: ButtonProps) {
```

리엑트 공식 홈페이지 코드와 넥스트 공식 코드가 props 에서 구조 분해를 하고 있어서 해당 방식 선호합니다 ㅎㅎ
물론 컴포넌트 내부 구현에서 구조 분해하는 라이브러리도 충분히 많이 있는 것 같네요